test/extensions/tracers/xray/tracer_test.cc

```diff
@@ -38,15 +38,16 @@ struct MockDaemonBroker : DaemonBroker {
 struct TraceProperties {
   TraceProperties(const std::string span_name, const std::string origin_name,
                   const std::string aws_key_value, const std::string operation_name,
-                  const std::string http_method, const std::string http_url,
-                  const std::string user_agent)
+                  const std::string direction, const std::string http_method,
```

Comment:
What influenced you to insert `direction` before the existing parameter order?  I would have expected the new parameter at the end so that existing callers don't accidentally mix parameters since these are all strings.

Reply:
Original idea was to keep `operation_name` & `direction` arguments next to each other.
Now I moved this to the last, so that in future if we keep introducing any more arguments we will continue to introduce at the end.